src/common/camera_control.c

```diff
@@ -824,8 +841,67 @@ static void dt_camctl_detect_cameras(const dt_camctl_t *c)
     } while(citem && (citem = g_list_next(citem)) != NULL);
   }
 
+  /* check c->locked_cameras in available_cameras */
+  if(c->locked_cameras && g_list_length(c->locked_cameras) > 0)
+  {
+    GList *c_lock_item = c->locked_cameras;
+    do
+    {
+      dt_camera_locked_t *locked_cam = (dt_camera_locked_t *)c_lock_item->data;
+      gboolean remove_cam = TRUE;
+      for(int i = 0; i < gp_list_count(available_cameras); i++)
+      {
+        const gchar *mymodel;
+        const gchar *myport;
+        gp_list_get_name(available_cameras, i, &mymodel);
+        gp_list_get_value(available_cameras, i, &myport);
+        if((g_strcmp0(mymodel, locked_cam->model) == 0) && (g_strcmp0(myport, locked_cam->port) == 0))
+          remove_cam = FALSE;
+      }
+      if(remove_cam)
+      {
+        dt_print(DT_DEBUG_CAMCTL, "[camera_control] move device %s on port %s from locked camera list to camera list as it is available\n",
+                 locked_cam->model, locked_cam->port);
+        /* remove camera from cached list.. */
+        dt_camera_locked_t *oldcam = (dt_camera_locked_t *)c_lock_item->data;
+        camctl->locked_cameras = c_lock_item = g_list_delete_link(c->locked_cameras, c_lock_item);
+        dt_camctl_locked_camera_destroy(oldcam);
+        changed_camera = TRUE;
+      }
+    } while(c_lock_item && (c_lock_item = g_list_next(c_lock_item)) != NULL);
+  }
+ 
   gp_list_unref(available_cameras);
 
+  /* check c->cameras in locked_cameras */
+  if((c->cameras && g_list_length(c->cameras) > 0) &&
+     (c->locked_cameras && g_list_length(c->locked_cameras) > 0))
```

Comment:
i think `g_list_length > 0` is pointless if `c->cameras` is not null, since non-null list is always at least 1 item long, right?

Reply:
ok